sampler.go

```diff
@@ -286,33 +328,30 @@ func (s *adaptiveSampler) Close() {
 }
```

Comment:
in line 318 we must protect the map against growing uncontrollably. I.e. we only add new samplers up to a certian number of unique ops. After that we should use some default sampler.